src/shims/unix/freebsd/foreign_items.rs

```diff
@@ -26,17 +26,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
                 this.write_null(dest)?;
             }
-            "pthread_set_name_np" => {
+            // Old vs new name
+            "pthread_set_name_np" | "pthread_setname_np" => {
                 let [thread, name] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
-                let max_len = usize::MAX; // freebsd does not seem to have a limit.
+                let max_len = 20; // freebsd silently truncates to MAXCOMLEN (+ 1).
                 let res = this.pthread_setname_np(
                     this.read_scalar(thread)?,
                     this.read_scalar(name)?,
                     max_len,
                 )?;
                 this.write_scalar(res, dest)?;
```

Comment:
I'm very surprised that this does not explode. Something is weird. This function returns nothing, and here we are assuming it returns an `i32`, so that should fail somewhere...